### PR TITLE
Misc. smart contract changes

### PR DESCRIPTION
**Pull request by wkaluza**

### CODE REVIEW

File: libs/http/src/response.cpp [hunk 1]
```diff
@@ -199,7 +199,7 @@ bool HTTPResponse::ParseHeaderLine(std::size_t line_idx, char const *begin, char
     return false;
   }
 
-  // construct the stings
+  // construct the strings
```
Comment (ejfitzgerald) [inline libs/http/src/response.cpp:202]: 🐝 🐝 🐝  - 😁

File: libs/ledger/include/ledger/dag/dag.hpp [hunk 2]
```diff
@@ -162,21 +161,21 @@ class DAG : public DAGInterface
   std::unordered_map<NodeHash, std::vector<DAGNodePtr>> loose_nodes_lookup_;  // nodes that are missing one or more references (waiting on NodeHash)
   // clang-format on
 
-  // std::unordered_map<NodeHash, uint64_t>                loose_nodes_ttl_;   // TODO(HUT): loose
-  // nodes management scheme
+  // TODO(HUT): loose nodes management scheme
```
Comment (ejfitzgerald) [inline libs/ledger/include/ledger/dag/dag.hpp]: Can you raise a tracking github issue for this

File: libs/ledger/src/transaction_processor.cpp [hunk 3]
```diff
@@ -33,6 +33,10 @@
 namespace fetch {
 namespace ledger {
 
+namespace {
+static constexpr char const *LOGGING_NAME = "TransactionProcessor";
```
Comment (ejfitzgerald) [inline libs/ledger/src/transaction_processor.cpp]: Pedantic - `static` not required

**Review by ejfitzgerald — CHANGES_REQUESTED** [review]
Looks good just a couple of minor things but otherwise lets try and get this in soon

File: libs/ledger/include/ledger/dag/dag.hpp (hunk 2 above)
Comment (wkaluza) [reply libs/ledger/include/ledger/dag/dag.hpp]: done #1642

File: libs/ledger/src/transaction_processor.cpp (hunk 3 above)
Comment (wkaluza) [reply libs/ledger/src/transaction_processor.cpp]: Pedantic is good

**Review by ejfitzgerald — APPROVED** [review]

**Pull request merged**